common/src/main/java/org/jboss/pnc/common/Configuration.java

```diff
@@ -33,19 +37,25 @@
      * @return Loaded configuration
      * @throws ConfigurationParseException Thrown if configuration file couldn't be loaded or parsed
      */
-    public T getModuleConfig(Class<T> moduleClass) throws ConfigurationParseException {
+    @SuppressWarnings("unchecked")
+    public <T extends AbstractModuleConfig> T getModuleConfig(Class<T> moduleClass) throws ConfigurationParseException {
```

Comment:
@matejonnet @rnc @slaskawi So add synchronized to the method and volatile to the cache field?